lib/tests/streamlit/util_test.py

```diff
@@ -185,3 +187,26 @@ def test_calc_md5_can_handle_bytes_and_strings(self):
             util.calc_md5("eventually bytes"),
             util.calc_md5("eventually bytes".encode("utf-8")),
         )
+
+    def test_timed_cleanup_cache_gc(self):
+        """Test that the TimedCleanupCache does not leave behind tasks when
+        the cache is not externally reachable"""
+        loop = asyncio.new_event_loop()
+        asyncio.set_event_loop(loop)
+
+        async def create_cache():
+            cache = util.TimedCleanupCache(maxsize=2, ttl=10)
+            cache["foo"] = "bar"
```

Comment:
Would it be possible to add another test to check that it actually gets removed from cache once it is expired? The downside would be that we need to wait at least the sleep time + the TTL time, so not sure if this is a good idea.

Reply:
I think I'd prefer to avoid adding a long test or changing the code to enable testing this.